File: src/main/interfaces/LicenseWindow.java
package main.interfaces;

import java.awt.Component;
import java.awt.Dimension;
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.net.URL;

import javax.swing.Box;
import javax.swing.BoxLayout;
import javax.swing.ImageIcon;
import javax.swing.JButton;
import javax.swing.JFrame;
import javax.swing.JPanel;
import javax.swing.WindowConstants;
import javax.swing.text.SimpleAttributeSet;
import javax.swing.text.StyleConstants;
import javax.swing.text.StyledDocument;
import java.awt.event.ActionEvent;
import java.awt.event.ActionListener;

public class LicenseWindow implements Runnable, ActionListener {
  private final JFrame frame;
  private final JButton agree, disagree;
  public boolean proceed = false;

  public LicenseWindow(int firstTime) throws IOException {

    URL url = getClass().getResource("/license_icon.png");
    assert url != null;
    ImageIcon icon = new ImageIcon(url);
    frame = new JFrame("Music Player | License");
    frame.setIconImage(icon.getImage());
    frame.setSize(500, 600);
    frame.setResizable(false);
    frame.setLocationRelativeTo(null);
    frame.setDefaultCloseOperation(WindowConstants.DISPOSE_ON_CLOSE);

    javax.swing.JLabel label = new javax.swing.JLabel("Music Player - License");
    label.setIcon(icon);
    label.setAlignmentX(Component.CENTER_ALIGNMENT);

    javax.swing.JTextPane textArea = new javax.swing.JTextPane();
    textArea.setEditable(false);

    BufferedReader br = new BufferedReader(
        new InputStreamReader(java.util.Objects.requireNonNull(getClass().getResource("/license.txt")).openStream()));
    StringBuilder license = new StringBuilder();
    try {
      String line = br.readLine();
      while (line != null) {
        license.append(line).append("\n");
        line = br.readLine();
      }
    } catch (Exception e) {
      e.printStackTrace();
    }

    textArea.setText(license.toString());
    StyledDocument doc = textArea.getStyledDocument();
    SimpleAttributeSet center = new SimpleAttributeSet();
    StyleConstants.setAlignment(center, StyleConstants.ALIGN_CENTER);
    doc.setParagraphAttributes(0, doc.getLength(), center, false);
    SimpleAttributeSet bold = new SimpleAttributeSet();
    StyleConstants.setBold(bold, true);
    doc.setCharacterAttributes(0, doc.getLength(), bold, false);
    textArea.setSize(new Dimension(400, 400));
    textArea.setFont(textArea.getFont().deriveFont(textArea.getFont().getSize() * 1f));
    textArea.setAlignmentX(Component.CENTER_ALIGNMENT);
    javax.swing.JScrollPane scrollPane = new javax.swing.JScrollPane(textArea);
    scrollPane.setBounds(0, 0, 150, 500);
    scrollPane.setAlignmentX(Component.CENTER_ALIGNMENT);
    scrollPane.setPreferredSize(new Dimension(450, 500));

    agree = new JButton("Agree");
    disagree = new JButton("Disagree");

    if (firstTime == -1) {
      JPanel panel = new JPanel();
      panel.setLayout(new BoxLayout(panel, BoxLayout.Y_AXIS));
      
      agree.addActionListener(this);
      agree.setAlignmentX(Component.CENTER_ALIGNMENT);

      
      disagree.addActionListener(this);
      disagree.setAlignmentX(Component.CENTER_ALIGNMENT);

      panel.add(agree);
      panel.add(disagree);

      frame.add(label);
      frame.add(Box.createHorizontalStrut(2));
      frame.add(Box.createVerticalStrut(7));
    }
    frame.add(scrollPane);

  }

  @Override
  public void run() {
    frame.pack();
    frame.setVisible(true);
  }

  /**
   * @param args
   * @throws IOException
   */
  public static void main(String[] args) throws IOException {
    new LicenseWindow(1).run();
  }

  @Override
  public void actionPerformed(ActionEvent e) {
    if(e.getSource() == agree) {
      frame.dispose();
      proceed = true;
    } else if(e.getSource() == disagree) {
      frame.dispose();
      proceed = false;
    }
  }

  public boolean isVisible() {
    return frame.isVisible();
  }
}

File: kt/app/interfaces/LicenseWindow.kt
package app.interfaces

import app.core.PropertiesReader.getVal
import app.interfaces.DocumentationWindow.run
import app.core.JSONParser.parseElement
import app.core.Cache.cleanCache
import app.core.PropertiesReader.Companion.reset
import app.interfaces.theme.LAFCommitter.setMultTheme
import app.interfaces.theme.Parser.getTheme
import app.interfaces.theme.Parser.parseThemeToProperty
import app.core.Host.Companion.openFileBrowser
import app.core.LifePreserver.saveToPrevDir
import java.lang.InterruptedException
import app.interfaces.dialog.ErrorMessage
import java.awt.event.ActionListener
import javax.swing.event.ChangeListener
import java.lang.Runnable
import backend.audioutil.Player
import app.interfaces.WindowPanel
import java.lang.Math
import kotlin.jvm.Synchronized
import java.awt.event.ActionEvent
import app.interfaces.dialog.FrameConfirmDialog
import app.interfaces.SelectFileWindow
import javax.swing.event.ChangeEvent
import app.interfaces.event.FrameOrganizer
import kotlin.Throws
import java.io.IOException
import kotlin.jvm.JvmStatic
import app.interfaces.LicenseWindow
import java.io.BufferedReader
import java.lang.StringBuilder
import javax.swing.text.StyledDocument
import javax.swing.text.SimpleAttributeSet
import javax.swing.text.StyleConstants
import app.CLI
import app.interfaces.WelcomeWindow
import app.interfaces.SettingsWindow
import app.interfaces.DocumentationWindow
import app.interfaces.event.WebsiteButtons
import app.core.PropertiesReader
import java.awt.event.ItemListener
import app.telemetry.FileIntegrity
import app.interfaces.dialog.OKWindow
import java.awt.event.ItemEvent
import app.interfaces.theme.LAFCommitter
import app.core.Host
import app.core.LifePreserver
import java.awt.BorderLayout
import java.awt.Color
import java.awt.Component
import java.awt.Dimension
import java.io.InputStreamReader
import java.lang.Exception
import java.util.*
import javax.swing.*

class LicenseWindow(firstTime: Int) : Runnable, ActionListener {
    private val frame: JFrame
    private val agree: JButton
    private val disagree: JButton
    var proceed = false
    override fun run() {
        frame.pack()
        frame.isVisible = true
    }

    override fun actionPerformed(e: ActionEvent) {
        if (e.source === agree) {
            frame.dispose()
            proceed = true
        } else if (e.source === disagree) {
            frame.dispose()
            proceed = false
        }
    }

    companion object {
        /**
         * @param args
         * @throws IOException
         */
        @Throws(IOException::class)
        @JvmStatic
        fun main(args: Array<String>) {
            LicenseWindow(1).run()
        }
    }

    init {
        val url = javaClass.getResource("/icons/others/license_icon.png")!!
        val icon = ImageIcon(url)
        frame = JFrame("Music Player | License")
        frame.iconImage = icon.image
        frame.setSize(500, 600)
        frame.isResizable = false
        frame.setLocationRelativeTo(null)
        frame.defaultCloseOperation = WindowConstants.DISPOSE_ON_CLOSE
        val label = JLabel("Music Player - License")
        label.icon = icon
        label.alignmentX = Component.CENTER_ALIGNMENT
        val textArea = JTextPane()
        textArea.isEditable = false
        val br = BufferedReader(
                InputStreamReader(Objects.requireNonNull(javaClass.getResource("/license.txt")).openStream()))
        val license = StringBuilder()
        try {
            var line = br.readLine()
            while (line != null) {
                license.append(line).append("\n")
                line = br.readLine()
            }
        } catch (e: Exception) {
            e.printStackTrace()
        }
        textArea.text = license.toString()
        val doc = textArea.styledDocument
        val center = SimpleAttributeSet()
        StyleConstants.setAlignment(center, StyleConstants.ALIGN_CENTER)
        doc.setParagraphAttributes(0, doc.length, center, false)
        val bold = SimpleAttributeSet()
        StyleConstants.setBold(bold, true)
        doc.setCharacterAttributes(0, doc.length, bold, false)
        textArea.size = Dimension(400, 400)
        textArea.font = textArea.font.deriveFont(textArea.font.size * 1f)
        textArea.alignmentX = Component.CENTER_ALIGNMENT
        val scrollPane = JScrollPane(textArea)
        scrollPane.setBounds(0, 0, 150, 500)
        scrollPane.alignmentX = Component.CENTER_ALIGNMENT
        scrollPane.preferredSize = Dimension(450, 500)
        agree = JButton("Agree")
        disagree = JButton("Disagree")
        if (firstTime == -1) {
            val panel = JPanel()
            panel.layout = BoxLayout(panel, BoxLayout.Y_AXIS)
            agree.addActionListener(this)
            agree.alignmentX = Component.CENTER_ALIGNMENT
            disagree.addActionListener(this)
            disagree.alignmentX = Component.CENTER_ALIGNMENT
            panel.add(agree)
            panel.add(disagree)
            frame.add(label)
            frame.add(Box.createHorizontalStrut(2))
            frame.add(Box.createVerticalStrut(7))
        }
        frame.add(scrollPane)
        frame.isUndecorated = true
        frame.addComponentListener(FrameOrganizer(frame))
    }
}